raphtory/src/graph_window.rs

```diff
@@ -193,75 +193,123 @@ impl WindowedVertex {
             .collect::<HashMap<String, Vec<(i64, Prop)>>>()
     }
 
```

Comment:
For all of these methods, it would be much better to match on the option and call the appropriate variant i.e. _window or not rather than unwrapping with a default window. The windowed variants of the methods will likely always have some overhead!